src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`2.IVsLanguageBlock.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Host;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Extensions;
+using Microsoft.VisualStudio.Text;
+using Roslyn.Utilities;
+
+using IVsLanguageBlock = Microsoft.VisualStudio.TextManager.Interop.IVsLanguageBlock;
+using IVsTextLines = Microsoft.VisualStudio.TextManager.Interop.IVsTextLines;
+using VsTextSpan = Microsoft.VisualStudio.TextManager.Interop.TextSpan;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
+{
+    internal abstract partial class AbstractLanguageService<TPackage, TLanguageService>
+        : IVsLanguageBlock
+    {
+        public int GetCurrentBlock(
+            IVsTextLines pTextLines,
+            int iCurrentLine,
+            int iCurrentChar,
+            VsTextSpan[] ptsBlockSpan,
+            out string pbstrDescription,
+            out int pfBlockAvailable)
+        {
+            var foundBlock = false;
+            string description = null;
+            var span = default(TextSpan);
+
+            var snapshot = this.EditorAdaptersFactoryService.GetDataBuffer(pTextLines).CurrentSnapshot;
+            var position = snapshot.GetPosition(iCurrentLine, iCurrentChar);
+
+            var waitIndicator = this.Package.ComponentModel.GetService<IWaitIndicator>();
+            waitIndicator.Wait(
+                ServicesVSResources.CurrentBlock,
+                ServicesVSResources.DeterminingCurrentBlock,
+                allowCancel: true,
+                action: context =>
+                {
+                    foundBlock = VsLanguageBlock.GetCurrentBlock(snapshot, position, context.CancellationToken, ref description, ref span);
```

Comment:
I tested adding a `throw new OperationCanceledException();` here, and the find dialog handled it the same as returning nothing found (doesn't show the current block scope as an option).  I'm inclined to leave the code as is.